config/parse.go

```diff
@@ -134,80 +96,125 @@ func (ctx *parseContext) isSnippet(name string) (bool, string) {
 	return false, ""
 }
 
-func (ctx *parseContext) readCfgNodes() ([]Node, error) {
+// readNodes reads nodes from currently parsed block.
+//
+// Lexer's cursor should point to opening brace
+// name arg0 arg1 {  #< this one
+//   c0
+//   c1
+// }
+//
+// To stay consistent with readNode after this function returns lexer's cursor pointer
+// to the last token of the black (closing brace).
+func (ctx *parseContext) readNodes() ([]Node, error) {
 	res := []Node{}
-	if ctx.parens > 255 {
-		return res, ctx.Err("Nesting limit reached")
+
+	// Refuse to continue is nesting is too big.
+	if ctx.nesting > 255 {
+		return res, ctx.Err("nesting limit reached")
 	}
+
+	ctx.nesting++
+
 	for ctx.Next() {
+		// name arg0 arg1 {
+		//   c0
+		//   c1
+		// }
+		// ^ called when we hit } on separate line,
+		// This means block we are supposed to parse ended and we should stop.
```

Comment:
"we are supposed to parse ended"

Typo here

Reply:
Fixed (64ef51c).